### DISCUSSION TITLE
Printing always selects the xs layout unless scaled down in printing dialogue

### DISCUSSION THREAD

**Discussion by FelixDKBinder** [quasarframework/quasar#9627] [General - Components / Directives / etc]
Hi!

I'm working on a project of digitizing paper forms into responsive forms in an application. I've used a responsive layout ranging from mobile (xs) up to desktop (xl) and everything works like a charm. 

A problem surges when trying to print the form into A4 paper size. The original A4 form was replicated into the desktop layout and the digital form should have the same layout as the original when printed, but the print preview shows the mobile layout instead. This happens independently of the actual screen layout. A minimal example for a Single File Component is given below:

    <template>
      <div class="row">
          <q-input class="col-12 col-md-3" label="Input1"></q-input>
          <q-input class="col-12 col-md-3" label="Input2"></q-input>
          <q-input class="col-12 col-md-3" label="Input3"></q-input>
          <q-input class="col-12 col-md-3" label="Input4"></q-input>
      </div>
    </template>

I have noticed also that there are different print previews on Chrome and Firefox if using col-sm-3 instead.

To fix this problem and printing the correct layout I've tried modifying the width of many elements (including the body outside of the scope of the SFC and the @page size) either normally or with @media print, sadly without success.
Through random playing around in the print dialogue I've found that setting the print scale to 50% modified the responsive layout to bigger resolutions. In my case even reaching the lg breakpoint and resulting in the original form layout.

**Although I've found a manual solution to this problem, I would like to know if there is any way to automatically set for which screen size the HTML should be printed. At the very least something that makes the scaling from the print dialogue not influence the layout.** 

There is also the possibility of recreating the form with the printing related visibility classes, but doing so would mean duplicating the whole forms for something that may be solved some other way and will be my last case scenario.

Thank your very much for your time and any possible answers!

**Comment by ontwikkelfabriek** [comment]
@FelixDKBinder 

Did you see the printing classes in Quasar:
https://quasar.dev/style/visibility#printing-related

And this `vue-html-to-paper` repo for printing in Vue:
https://randomcodetips.com/vue-html-to-paper/

The github repo of `vue-html-to-paper` :
https://github.com/mycurelabs/vue-html-to-paper

**Reply by FelixDKBinder** [reply]
Thank your for your answer! As I see in the index.js, vue-html-to-paper simply facilitates calling the print dialogue the same way [Print.js](https://printjs.crabbly.com/) does and this one hasn't helped with the layout either.

I did see the printing classes in Quasar and if I were to use them in my example, I would have to create a duplicate for each component in the form like:

    <q-input class="print-hide col-12 col-md-3" label="Input1"></q-input>
    <q-input class="print-only col-12" label="Input1"></q-input>

I know it is possible to create [class bindings](https://vuejs.org/v2/guide/class-and-style.html), but I would need to look for the variable responsible for switching between print-hide and print-only (which I didn't have much time to do yet) and in my special case it would need to be inline instead of using the script part of the SFC.
